### PR TITLE
Registry setting for pipeline modification.

### PR DESCRIPTION
**Pull request by michalpelka**
## What does this PR do?

Introduce registry setting to enable pipeline modification for the camera sensor.

## How was this PR tested?

run against roscon2023 demo.

### CODE REVIEW

**Review by pijaro — APPROVED** [review]

File: Gems/ROS2/Code/Source/Camera/CameraSensor.cpp [hunk 1]
```diff
@@ -27,6 +28,9 @@ namespace ROS2
 {
     namespace Internal
     {
+        constexpr AZStd::string_view AllowCameraPipelineModificationKey = "/O3DE/ROS2/Camera/AllowPipelineModification";
+
+
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/Camera/CameraSensor.cpp:33]:
```suggestion
```

**Review by jhanca-robotecai — APPROVED** [review]
I am fine with merging the change (target `development`), but due to the changes in project templates, we will have to bump their versions, redo the `repo.json` file, and release them when cherry-picking to the `point-release` branch.

File: Gems/ROS2/Code/Source/Camera/CameraSensor.cpp [hunk 2]
```diff
@@ -107,7 +111,18 @@ namespace ROS2
...
     {
-        AZ_TracePrintf("CameraSensor", "Initializing pipeline for %s\n", m_cameraSensorDescription.m_cameraName.c_str());
+        bool allowModification = false;
+        auto* registry = AZ::SettingsRegistry::Get();
+        AZ_Assert(registry, "No Registry available");
+        if (registry)
+        {
+            registry->Get(allowModification, Internal::AllowCameraPipelineModificationKey);
+        }
+        AZ_TracePrintf(
```
Comment (spham-amzn) [inline Gems/ROS2/Code/Source/Camera/CameraSensor.cpp:121]: `AZ_TracePrintf` is deprecated, use `AZ_Trace` instead
```suggestion
        AZ_Trace(
```

File: Gems/ROS2/Code/Source/Camera/CameraSensor.cpp [hunk 3]
```diff
@@ -107,7 +111,18 @@ namespace ROS2
...
+            "CameraSensor",
+            "Initializing pipeline for %s, pipeline modification : %d\n",
+            m_cameraSensorDescription.m_cameraName.c_str(),
+            allowModification);
```
Comment (spham-amzn) [inline Gems/ROS2/Code/Source/Camera/CameraSensor.cpp:125]:
```suggestion
            "Initializing pipeline for %s, pipeline modification : %s\n",
            m_cameraSensorDescription.m_cameraName.c_str(),
            allowModification?"yes":"no");
```

File: Templates/Ros2FleetRobotTemplate/Template/Registry/ros2.setreg [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        },
+        "ROS2":
+        {
+            "SteadyClock" : true,
+            "Camera":
+            {
+                "AllowPipelineModification": true
+            }
+        }
+
```
Comment (spham-amzn) [inline Templates/Ros2FleetRobotTemplate/Template/Registry/ros2.setreg:19]:
```suggestion
```

File: Templates/Ros2RoboticManipulationTemplate/Template/Registry/ros2.setreg [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        },
+        "ROS2":
+        {
+            "SteadyClock" : true,
+            "Camera":
+            {
+                "AllowPipelineModification": true
+            }
+        }
+
```
Comment (spham-amzn) [inline Templates/Ros2RoboticManipulationTemplate/Template/Registry/ros2.setreg:19]:
```suggestion
```

File: Templates/Ros2FleetRobotTemplate/template.json
Comment (spham-amzn) [inline Templates/Ros2FleetRobotTemplate/template.json:1]: As @jhanca-robotecai mentioned, when updating project templates, we need to bump the version number at the top of this json file as well

File: Templates/Ros2ProjectTemplate/template.json
Comment (spham-amzn) [inline Templates/Ros2ProjectTemplate/template.json:1]: Same version bump for this file

**Pull request merged**